src/Microsoft.TemplateEngine.Utils/TemplateParameter.cs

```diff
@@ -100,8 +113,12 @@ public TemplateParameter(JObject jObject)
         [JsonProperty]
         public string Name { get; }
 
-        [JsonProperty]
-        public TemplateParameterPriority Priority { get; }
+        [JsonIgnore]
+        [Obsolete("Use Precedence instead.")]
+        public TemplateParameterPriority Priority => Precedence.PrecedenceDefinition.ToTemplateParameterPriority();
+
+        [JsonIgnore]
```

Comment:
I think this is better handled via custom converter instead.
https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_JsonPropertyAttribute.htm 
`JsonPropertyAttribute` has a way to specify converter to be used https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonPropertyAttribute_ItemConverterType.htm
The converter implementation itself may live in `JExtensions` to be reused here and in search.

Reply:
Not sure if I fully agree. Custom converter - I'd need to handle serialization/deserialization in custom way - while here I just have a type that marks the properties (plus custom converter cannot cirmuvent a need for converting legacy and new tag to a single property).
If logic reusal will be needed, then the serializable `TemplateParameterPrecedence` type can be shared similarly as `JExtensions`.

Reply:
Let's sync up on this one additionally. I feel that this additional implementation might be not even needed as all public properties will be serialized anyway.